### PR TITLE
Return mewt wrapped instances

### PR DESCRIPTION
**Pull request by bali182**
This PR changes the following:
- Code style - since we don't need to be so careful about the size now, I'm using `const` where possible and moved stuff to separate lines to be more readable
- All non-mutating array methods skip the cloning process before wrapped in `mewt`
- All mutating array methods do the cloning before wrapped in `mewt`
- All double return value methods wrap their second argument (cloned array) in mewt
- `$set` and `$unset` wrap in `mewt`
- Added tests for all of this stuff

@sdgluck let me know what you think!

### CODE REVIEW

**Comment by bali182** [comment]
Not really sure what you think about that everything is `const` and stuff is moved to newlines... Let me know if you prefer the way it is currently

File: index.js [hunk 1]
```diff
@@ -1,30 +1,51 @@
+const multiRet = ['push', 'pop', 'shift', 'unshift']
```
Comment (sdgluck) [inline index.js]: I appreciate the motivation behind the change to `const`, however would prefer to keep using `let` simply to keep the size down (part of writing this lib was a challenge as to how small I could make it!).

**Comment by bali182** [comment]
Minified size grew quite a bit its 778 bytes now 😢

File: test.js [hunk 2]
```diff
@@ -3,6 +3,12 @@
 const test = require('tape')
 const mewt = require('./lib')
 
+const createAssertMewt = t => input => {
+  t.equal(typeof input.$set, 'function')
+  t.equal(typeof input.$unset, 'function')
+  // TODO any other way to test? Cannot check if instanceof Proxy
```
Comment (sdgluck) [inline test.js]: I think this is an adequate way to test if we are looking at a `mewt`, so no need for a TODO here.

**Comment by sdgluck** [comment]
For some reason the variables defined outside of the `mewt` function are not being minified. I don't see why that would happen though. Confusing...

File: index.js (hunk 1 above)
Comment (bali182) [reply index.js]: Completely respect that, will change if you say you are sure. But we gained 10 bytes in the minified output when I replaced all `const`s with `let`s. Are you sure you want this?

File: index.js [hunk 3]
```diff
@@ -1,30 +1,51 @@
...
+const mutArrMethods = ['reverse', 'sort', 'splice', 'fill', 'copyWithin']
```
Comment (sdgluck) [inline index.js]: Making each of these collections of method names a space-delimited string would save quite a few bytes I think. You can use the `String#includes` method as is done currently.

Comment (bali182) [reply index.js]: Sure, it's a good idea! And I think this way it's OK to move these back to the method. Even if the interpreter is not smart enough and allocates 3 strings each time you wrap something into a `mewt` (I don't think its that dumb though) it's not as bad as having 3 arrays created each time :)

Comment (bali182) [reply index.js]: It's 657 bytes with `let`s and strings instead of arrays.

File: index.js (hunk 1 above)
Comment (sdgluck) [reply index.js]: That's strange! Not what I would expect at all. Maybe it is down to babili being in beta. If that's the case then we can leave the change to `const` as-is. 👍

File: index.js (hunk 3 above)
Comment (sdgluck) [reply index.js]: Cool, that's a decent reduction. Can always fine-tune the size later too.

File: index.js (hunk 1 above)
Comment (bali182) [reply index.js]: Results after changed strings to array:
`let`: 657 bytes
`const`: 665 bytes

Not sure how it does it but I won't sift through the minified output :D

If you say it's OK I'll leave it as `const`

Comment (sdgluck) [reply index.js]: Yep no problem

**Comment by sdgluck** [comment]
Ready for this to be merged? @bali182

**Comment by bali182** [comment]
I think it is!

**Comment by sdgluck** [comment]
Cheers! 🥇

**Pull request merged**